include/nbl/ext/ImGui/ImGui.h

```diff
@@ -9,124 +9,174 @@ namespace nbl::ext::imgui
-		bool render(nbl::video::IGPUCommandBuffer* const commandBuffer, nbl::video::ISemaphore::SWaitInfo waitInfo, const std::span<const VkRect2D> scissors = {});
+		bool render(video::IGPUCommandBuffer* const commandBuffer, video::ISemaphore::SWaitInfo waitInfo, const std::chrono::steady_clock::time_point waitPoint = std::chrono::steady_clock::now() + std::chrono::milliseconds(1u), const std::span<const VkRect2D> scissors = {});
 
-		//! registers lambda listener in which ImGUI calls should be recorded
+		//! registers lambda listener in which ImGUI calls should be recorded, use the returned id to unregister the listener
 		size_t registerListener(std::function<void()> const& listener);
+
+		//! unregisters listener with the given id
 		std::optional<size_t> unregisterListener(size_t id);
 
 		//! sets ImGUI context, you are supposed to pass valid ImGuiContext* context
 		void setContext(void* imguiContext);
 
-		//! creation parametrs
-		inline const S_CREATION_PARAMETERS& getCreationParameters() const { return m_creationParams; }
+		//! creates default pipeline layout for the UI resources, "texturesCount" argument is textures descriptor binding's array size. Samplers are immutable and part of the created layout, SResourceParameters::DefaultSamplerIx::COUNT is the size of the samplers descriptor binding's array
+		static core::smart_refctd_ptr<video::IGPUPipelineLayout> createDefaultPipelineLayout(video::IUtilities* const utilities, const SResourceParameters::SBindingInfo texturesInfo, const SResourceParameters::SBindingInfo samplersInfo, uint32_t texturesCount = 0x45);
+
+		//! mounts the extension's archive to given system - useful if you want to create your own shaders with common header included
+		static const core::smart_refctd_ptr<system::IFileArchive> mount(core::smart_refctd_ptr<system::ILogger> logger, system::ISystem* system, const std::string_view archiveAlias = "");
+
+		//! creation cached parametrs
+		inline const SCachedCreationParams& getCreationParameters() const { return m_cachedCreationParams; }
 
 		//! ImGUI graphics pipeline
-		inline nbl::video::IGPUGraphicsPipeline* getPipeline() { return pipeline.get(); }
+		inline const video::IGPUGraphicsPipeline* getPipeline() const { return m_pipeline.get(); }
 
 		//! image view default font texture
-		inline nbl::video::IGPUImageView* getFontAtlasView() { return m_fontAtlasTexture.get(); }
+		//! TODO: we cannot expose immutable view of our default font texture because user MUST be able to write a descriptor, 
+		//! we can have mutable getter or we can decide to not create any default font texture at all and force users 
+		//! to do it externally (have a static helper to do it which gives you mutable view)
+		inline video::IGPUImageView* getFontAtlasView() const { return m_fontAtlasTexture.get(); }
```

Comment:
remove the TODO, in this instance a non const pointer to an ImageView is fine, because you can't really do that much with an `IGPUImageView` anyway XD

Reply:
resolved